oct/tournament/views.py

```diff
@@ -36,6 +41,68 @@
 def error_500(request):
     return render(request, "tournament/error_500.html")
 
+def generate_roles_dict(roles, testing=False):
+    role_list = list(map(lambda r: (r.name[0]+r.name[1:]).lower(), roles))
+
+    if testing == True:
+        return dict(map(lambda role: (role, True), ["host", "registered_player", "custom_mapper", "mappooler", "playtester", "streamer", "commentator", "referee"]))
+    return dict(map(lambda role: (role, role in role_list), ["host", "registered_player", "custom_mapper", "mappooler", "playtester", "streamer", "commentator", "referee"]))
```

Comment:
this entire function should be `return dict(map(lambda: r: (r.name.lower(), r in roles), USER_DISPLAY_ORDER))`

Reply:
including a change in another comment below

Reply:
I'll handle this one